### PR TITLE
SDIT-1898 Switch mock casenotes-api to real one (from Nomis)

### PR DESCRIPTION
**Pull request by steverendell**
Add extra fields described in MICN-128

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/casenotes/CaseNotesResource.kt [hunk 1]
```diff
@@ -280,6 +293,20 @@ data class CreateCaseNoteRequest(
...
 )
 
+@JsonInclude(JsonInclude.Include.NON_NULL)
+data class CaseNoteAmendment(
+  @Schema(description = "Free format text body of amendment")
+  val text: String,
+  @Schema(description = "Author login name")
+  val authorUsername: String,
+  @Schema(description = "Author STAFF_ID")
+  val authorUserId: Long?,
```
Comment (petergphillips) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/casenotes/CaseNotesResource.kt:303]: odd that this is `userId`?

File: src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/casenotes/CaseNotesService.kt [hunk 2]
```diff
@@ -108,13 +112,65 @@ class CaseNotesService(
...
+      }
+
+      val amendText = caseNoteText.slice(matchResult.range.last + 1..startOfNext)
+      val staff = staffUserAccountRepository.findByUsername(user)?.staff
+
+      CaseNoteAmendment(
+        text = amendText,
+        authorUsername = user,
+        authorUserId = staff?.id,
+        authorName = staff?.run { "$firstName $lastName" },
```
Comment (petergphillips) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/casenotes/CaseNotesService.kt:168]: do you not need to format the author's name - won't it come back in capitals?

**Review by petergphillips — APPROVED** [review]
LGTM

File: src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/casenotes/CaseNotesResource.kt (hunk 1 above)
Comment (steverendell) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/casenotes/CaseNotesResource.kt:303]: True, i took this from the dps team spec, but agree, will change it next time

File: src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/casenotes/CaseNotesService.kt [hunk 3]
```diff
@@ -108,13 +112,65 @@ class CaseNotesService(
...
+
+      val (user, date, time) = matchResult.destructured
+      val startOfNext = if (matchLastIndex > index) {
+        matchResults.elementAt(index + 1).range.first - 1
+      } else {
+        caseNoteText.length - 1
+      }
+
+      val amendText = caseNoteText.slice(matchResult.range.last + 1..startOfNext)
+      val staff = staffUserAccountRepository.findByUsername(user)?.staff
```
[the remark this answers is not in the document]
Comment (steverendell) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/casenotes/CaseNotesService.kt:162]: Yep, and I did wonder about caching but the percentage of casenotes which have amendments (esp > 1) is low

File: src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/casenotes/CaseNotesService.kt (hunk 2 above)
Comment (steverendell) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/casenotes/CaseNotesService.kt:168]: Good Q, :shrug .. Ill ask Dave W what he wants

**Pull request merged**